### PR TITLE
Cleanup for v0.14 release

### PR DESCRIPTION
**Pull request by CarloLucibello**
Some small doc changes and remove some v0.12 deprecations

### CODE REVIEW

File: docs/src/gpu.md [hunk 1]
```diff
@@ -1,11 +1,23 @@
 # GPU Support
 
-NVIDIA GPU support should work out of the box on systems with CUDA and CUDNN installed. For more details see the [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl) readme.
+Flux doesn't force a specific GPU backend and the corresponding package dependencies on the users. 
+Thanks to the [package extension mechanism](
+https://pkgdocs.julialang.org/v1/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)),
+Flux conditionally load GPU specific code once a GPU package is made available (e.g. through `using CUDA`).
```
Comment (darsnack) [inline docs/src/gpu.md:6]:
```suggestion
Flux conditionally loads GPU specific code once a GPU package is made available (e.g. through `using CUDA`).
```

File: docs/src/gpu.md [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+
+NVIDIA GPU support requires the packages `CUDA.jl` and `cuDNN.jl` to be installed in the environment. In the julia REPL, type `] add CUDA, cuDNN` to install them. For more details see the [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl) readme.
 
 AMD GPU support is available since Julia 1.9 on systems with ROCm and MIOpen installed. For more details refer to the [AMDGPU.jl](https://github.com/JuliaGPU/AMDGPU.jl) repository.
 
 Metal GPU acceleration is available on Apple Silicon hardware. For more details refer to the [Metal.jl](https://github.com/JuliaGPU/Metal.jl) repository. Metal support in Flux is experimental and many features are not yet available.
 
+In order to trigger GPU support in Flux, you need to call `using CUDA`, `using AMDGPU` or `using Metal`
+in your code. Notice that for CUDA, explicitely loading also `cuDNN` is not required, but the package has to be installed in the environment.
```
Comment (darsnack) [inline docs/src/gpu.md:15]:
```suggestion
in your code. Notice that for CUDA, explicitly loading `cuDNN` is not required, but the package has to be installed in the environment. 
```

File: docs/src/gpu.md [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
 AMD GPU support is available since Julia 1.9 on systems with ROCm and MIOpen installed. For more details refer to the [AMDGPU.jl](https://github.com/JuliaGPU/AMDGPU.jl) repository.
 
 Metal GPU acceleration is available on Apple Silicon hardware. For more details refer to the [Metal.jl](https://github.com/JuliaGPU/Metal.jl) repository. Metal support in Flux is experimental and many features are not yet available.
 
+In order to trigger GPU support in Flux, you need to call `using CUDA`, `using AMDGPU` or `using Metal`
+in your code. Notice that for CUDA, explicitely loading also `cuDNN` is not required, but the package has to be installed in the environment. 
+
+
+!!! compat "Flux ≤ 0.13"
+    Old versions of Flux automatically installed CUDA.jl to provide GPU support. Starting from Flux v0.14, CUDA.jl is not a dependency anymore and has to be installed manually.
```
Comment (darsnack) [inline docs/src/gpu.md:19]:
```suggestion
    Old versions of Flux automatically installed CUDA.jl to provide GPU support. Starting from Flux v0.14, CUDA.jl is no longer a dependency and has to be installed intentionally by the user.
```

**Review by darsnack — APPROVED** [review]

File: src/utils.jl [hunk 4]
```diff
@@ -47,11 +47,7 @@ rng_from_array(::AbstractArray) = default_rng_value()
 
 @non_differentiable rng_from_array(::Any)
 
-if VERSION >= v"1.7"
-  default_rng_value() = Random.default_rng()
-else
-  default_rng_value() = Random.GLOBAL_RNG
-end
+default_rng_value() = Random.GLOBAL_RNG
```
Comment (darsnack) [inline src/utils.jl:50]:
```suggestion
default_rng_value() = Random.default_rng()
```

**Review by darsnack — DISMISSED** [review]

File: src/utils.jl [hunk 5]
```diff
@@ -47,11 +47,7 @@ rng_from_array(::AbstractArray) = default_rng_value()
 
 @non_differentiable rng_from_array(::Any)
 
-if VERSION >= v"1.7"
-  default_rng_value() = Random.default_rng()
-else
-  default_rng_value() = Random.GLOBAL_RNG
-end
+default_rng_value() = Random.default_rng()
```
[the remark this answers is not in the document]
Comment (CarloLucibello) [reply src/utils.jl]: `default_rng_value` deprecated

**Comment by CarloLucibello** [comment]
> One other place that needs touching is the first line of this:

done

**Comment by CarloLucibello** [comment]
Good to go? need an approval

**Comment by CarloLucibello** [comment]
green lights

**Comment by CarloLucibello** [comment]
Need approval

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**